daemon/daemonbuilder/builder.go

```diff
@@ -29,12 +28,29 @@ import (
 
 // Docker implements builder.Docker for the docker Daemon object.
 type Docker struct {
-	Daemon      *daemon.Daemon
+	Daemon      Builder
 	OutOld      io.Writer
 	AuthConfigs map[string]cliconfig.AuthConfig
 	Archiver    *archive.Archiver
 }
 
+// Builder defines an interface that the responsible for building images must implement
+type Builder interface {
```

Comment:
I feel like this is half way to a duplicate of the `Docker` interface in `builder/`.